From 425bc6efcbb81bb3899b4ea0058e882843d2e933 Mon Sep 17 00:00:00 2001
From: Ricardo Branco <rbranco@suse.de>
Date: Sat, 28 Feb 2026 00:19:19 +0100
Subject: [PATCH] integration/image: pull ppc64le and s390x in
 TestSaveAndLoadPlatform

When savePlatforms is nil, ImageSave exports all platforms referenced
in the image index. On ppc64le and s390x hosts, this includes the
native platform which was never pulled, causing a "content digest not
found" error.

Pull ppc64le and s390x alongside the other platforms so their content
is available in the local store when the full-index save is attempted.

Signed-off-by: Ricardo Branco <rbranco@suse.de>
---
 integration/image/save_test.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/integration/image/save_test.go b/integration/image/save_test.go
index 82194f15170f7..cbe057e4c3229 100644
--- a/integration/image/save_test.go
+++ b/integration/image/save_test.go
@@ -235,19 +235,25 @@ func TestSaveAndLoadPlatform(t *testing.T) {
 			containerdStoreOnly: true,
 			pullPlatforms: []ocispec.Platform{
 				{OS: "linux", Architecture: "amd64"},
+				{OS: "linux", Architecture: "ppc64le"},
 				{OS: "linux", Architecture: "riscv64"},
+				{OS: "linux", Architecture: "s390x"},
 				{OS: "linux", Architecture: "arm64", Variant: "v8"},
 			},
 			savePlatforms: nil,
 			loadPlatforms: nil,
 			expectedSavedPlatforms: []ocispec.Platform{
 				{OS: "linux", Architecture: "amd64"},
+				{OS: "linux", Architecture: "ppc64le"},
 				{OS: "linux", Architecture: "riscv64"},
+				{OS: "linux", Architecture: "s390x"},
 				{OS: "linux", Architecture: "arm64", Variant: "v8"},
 			},
 			expectedLoadedPlatforms: []ocispec.Platform{
 				{OS: "linux", Architecture: "amd64"},
+				{OS: "linux", Architecture: "ppc64le"},
 				{OS: "linux", Architecture: "riscv64"},
+				{OS: "linux", Architecture: "s390x"},
 				{OS: "linux", Architecture: "arm64", Variant: "v8"},
 			},
 		},
